cmd/containerd-shim/main_unix.go

```diff
@@ -59,6 +60,9 @@ func init() {
 	flag.StringVar(&runtimeRootFlag, "runtime-root", proc.RuncRoot, "root directory for the runtime")
 	flag.StringVar(&criuFlag, "criu", "", "path to criu binary")
 	flag.BoolVar(&systemdCgroupFlag, "systemd-cgroup", false, "set runtime to use systemd-cgroup")
+	// currently, the `containerd publish` utility is embedded in the daemon binary for some reason.
+	// The daemon invokes ``containerd-shim -containerd-binary ...` with its own os.Executable() path.
+	flag.StringVar(&containerdBinaryFlag, "containerd-binary", "containerd", "path to containerd binary (used for `containerd publish`)")
```

Comment:
Shouldn't the default be the current executable path?

Reply:
no, because this is in the shim

Reply:
Oups, my bad, missed the path.